### ISSUE TITLE
Connector might flood the drive with gigabytes of logs on XMPP error

### ISSUE THREAD

**Issue by jacobmarble** [google/cups-connector#3]
When the server side of the XMPP conversation fails, "connection reset by peer" is logged zillions of times:
```
W0314 07:40:56.883629    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer
W0314 07:40:56.903180    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer
W0314 07:40:56.903200    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer
W0314 07:40:56.903215    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer
W0314 07:40:56.903229    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer
W0314 07:40:56.903243    8815 xmpp.go:177] Failed to read the next start element: read tcp 64.233.177.125:443: connection reset by peer

another server:
W0319 15:20:32.449441    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449528    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449580    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449632    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449683    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449733    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449784    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449835    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449885    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449936    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.449987    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450037    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450085    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450136    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450185    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450236    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450284    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450335    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450385    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450435    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450487    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450536    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450636    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450689    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450739    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450789    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450839    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450891    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450942    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.450993    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451044    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451095    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451146    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451195    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451245    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451299    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451349    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451399    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451452    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451503    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451551    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451602    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451652    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451700    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
W0319 15:20:32.451751    1691 xmpp.go:177] Failed to read the next start element: read tcp 74.125.28.125:443: connection reset by peer
```

**Comment by akorobkin** [comment]
Happened again yesterday with the connector build as of 7/17. 


Log file created at: 2015/07/29 16:04:19
Running on machine: printserver2
Binary: Built with gc go1.4.2 for linux/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
E0729 16:04:19.094438   30062 connector.go:32] Google Cloud Print CUPS Connector version 2015.07.17-linux
W0729 16:04:23.292133   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
E0729 16:04:38.622177   30062 connector.go:107] Ready to rock as proxy 'printserver2'
W0729 17:04:38.674670   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 18:04:38.747481   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 19:04:38.835427   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 20:04:38.895271   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 21:04:38.956080   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 22:04:39.009105   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0729 23:04:39.064220   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 00:04:39.153408   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 01:04:39.207202   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 02:04:39.260748   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 03:04:39.380716   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 04:04:39.433747   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 05:04:39.487888   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 06:04:39.541127   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 07:04:39.595038   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 08:04:39.649292   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 09:04:39.704260   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 10:04:39.757467   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 11:04:39.825248   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 12:04:39.879034   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 13:04:39.932475   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 14:04:39.986014   30062 cups.go:645] Received badly-formatted markers from CUPS: Black Cartridge HP CE250X;Cyan Cartridge HP CE251A;Magenta Cartridge HP CE253A;Yellow Cartridge HP CE252A;Image Fuser Kit HP 110V-CE484A; 220V-CE506A;Toner Collection Unit HP CE254A, toner-cartridge;toner-cartridge;toner-cartridge;toner-cartridge;fuser;cleaner-unit, 25;89;60;78;83;100
W0730 19:11:43.501703   30062 internal-xmpp.go:229] Unexpected element while waiting for print message: &{Name:{Space:http://etherx.jabber.org/streams Local:error} Attr:[]}
W0730 19:11:43.539812   30062 internal-xmpp.go:229] Unexpected element while waiting for print message: &{Name:{Space:urn:ietf:params:xml:ns:xmpp-streams Local:see-other-host} Attr:[{Name:{Space: Local:xmlns} Value:urn:ietf:params:xml:ns:xmpp-streams}]}
W0730 19:11:43.539861   30062 internal-xmpp.go:229] Unexpected element while waiting for print message: &{Name:{Space:urn:ietf:params:xml:ns:xmpp-streams Local:text} Attr:[{Name:{Space:xmlns Local:str} Value:urn:ietf:params:xml:ns:xmpp-streams}]}
E0730 19:12:57.379072   30062 xmpp.go:130] XMPP conversation died; restarting
E0730 21:03:48.055397   30062 xmpp.go:130] XMPP conversation died; restarting
W0730 21:50:14.935255   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935297   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935314   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935329   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935345   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935360   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935374   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935390   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935405   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935420   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935435   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
W0730 21:50:14.935450   30062 internal-xmpp.go:170] Failed to read the next start element: read tcp 74.125.71.125:443: connection timed out
<20Gb of logs skipped>

**Comment by jacobmarble** [comment]
Another one:
E0812 06:39:27.233730 08755 xmpp.go:130] XMPP conversation died; restarting
E0812 08:41:43.718194 08755 xmpp.go:130] XMPP conversation died; restarting
F0812 08:42:13.726229 08755 xmpp.go:132] Failed to keep XMPP conversation alive: While starting XMPP, failed to get access token (password): Post https://accounts.google.com/o/oauth2/token: dial tcp: i/o timeout
goroutine 1145 [running]:
github.com/golang/glog.stacks(0x965c0d00, 0x0, 0x0, 0x0)
        /home/pi/go/src/github.com/golang/glog/glog.go:726 +0xdc
github.com/golang/glog.(*loggingT).output(0x4a0ba0, 0x3, 0x9658ab40)
        /home/pi/go/src/github.com/golang/glog/glog.go:677 +0x22c
github.com/golang/glog.(*loggingT).printf(0x4a0ba0, 0x3, 0x373db8, 0x2a, 0x968c9f74, 0x1, 0x1)
        /home/pi/go/src/github.com/golang/glog/glog.go:635 +0x1a8
github.com/golang/glog.Fatalf(0x373db8, 0x2a, 0x968c9f74, 0x1, 0x1)
        /home/pi/go/src/github.com/golang/glog/glog.go:1033 +0x54
github.com/google/cups-connector/xmpp.(*XMPP).keepXMPPAlive(0x9659a1e0)
        /home/pi/go/src/github.com/google/cups-connector/xmpp/xmpp.go:132 +0x208
created by github.com/google/cups-connector/xmpp.(*XMPP).startXMPP
        /home/pi/go/src/github.com/google/cups-connector/xmpp/xmpp.go:113 +0x2a4

**Comment by jacobmarble** [comment]
My last comment is not the infinite log bug, but I'm leaving it because it's fixed in the same commit.

#45

**Comment by jacobmarble** [comment]
@akorobkin please close this issue after you are satisfied that this is fixed.

**Comment by riwsakiP8** [comment]
fix#45

**Comment by jacobmarble** [comment]
Fixed in #45
